astropy/coordinates/tests/test_sky_coord.py

```diff
@@ -610,6 +620,22 @@ def test_position_angle_directly():
     assert result.value == 0.
 
 
+def test_directional_offset_by():
```

Comment:
Done.  Multiple such tests added.

posang=0 is Due North -> increasing dec (until it goes over the pole)
posang=90deg  is Due East -> increasing RA.
Small and large separations, including =180 (antipode), >180, =360 (return to start), and >360deg